packages/core/clients/kernelAccountClient.ts

```diff
@@ -102,18 +109,28 @@ export const createKernelAccountClient = <
     })
 
     let middleware = parameters.middleware
-    if (
+
+    const isPimlicoProvider =
+        client.transport?.url && hasPimlicoAsProvider(client.transport.url)
+    const isPimlicoCli =
+        client.transport?.url && isPimlicoClient(client.transport.url)
+
+    const shouldUpdateMiddleware =
         (!middleware ||
             (typeof middleware !== "function" && !middleware.gasPrice)) &&
-        client.transport?.url &&
-        hasPimlicoAsProvider(client.transport.url)
-    ) {
-        const gasPrice = () => getUserOperationGasPrice(client)
+        (isPimlicoProvider || isPimlicoCli)
+
+    if (shouldUpdateMiddleware) {
+        const gasPrice = isPimlicoCli
```

Comment:
I think we don't need to handle calling `eth_getUserOperationGasPrice` explicitly if the user is using PimlicoClient explicitly. They should pass it manually as per permissionless docs.